### ISSUE TITLE
Transient `repository_files` failure during IT, drs_uri is None

### ISSUE THREAD

**Issue by dsotirho-ucsc** [DataBiosphere/azul#5757]
A transient IT failure wes experienced on `anvilprod`.

IT Job [17117](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/17117):

```
2023-12-05 19:31:19,938    INFO MainThread test.integration_test: Beginning sub-test [repository_files] {'catalog': 'anvil3-it'}
2023-12-05 19:31:19,939    INFO MainThread test.integration_test: Making GET request to 'https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/9cb83be7-2d48-48d7-b852-64bab0fd32ea?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z'
2023-12-05 19:31:19,939   DEBUG MainThread test.integration_test: … without request body
2023-12-05 19:31:20,276    INFO MainThread test.integration_test: Got 500 response after 0.336s from GET to https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/9cb83be7-2d48-48d7-b852-64bab0fd32ea?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z
2023-12-05 19:31:20,276   DEBUG MainThread test.integration_test: … with response headers HTTPHeaderDict({'Content-Type': 'text/plain', 'Content-Length': '1010', 'Connection': 'keep-alive', 'Date': 'Tue, 05 Dec 2023 19:31:20 GMT', 'x-amzn-RequestId': '263dcb2c-42fc-4fe0-b8fa-bfdc265d0be4', 'Access-Control-Allow-Origin': '*', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Access-Control-Allow-Headers': 'Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key', 'x-amz-apigw-id': 'PfAVSG5AoAMEI-w=', 'X-Amzn-Trace-Id': 'Root=1-656f7a87-35d76ae162dcbdf777e90273;Sampled=0;lineage=45061563:0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 4db130e87be66fce9731567ae0669c56.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'IAD89-C3', 'X-Amz-Cf-Id': 'gkjCdCFrddqOkAxJfHcHf6RryphoVHXf6aOZHrjeu_PCj35hETGnjw=='})
2023-12-05 19:31:20,276   DEBUG MainThread test.integration_test: … with response body b'Traceback (most recent call last):\n  File "/var/task/chalice/app.py", line 1914, in _get_view_function_response\n    response ...'
2023-12-05 19:31:20,276    INFO MainThread test.integration_test: Failed sub-test [repository_files] {'catalog': 'anvil3-it'}
  test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [repository_files] (catalog='anvil3-it') ... FAIL
 ```

[CloudWatch](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2023-12-05T19*3a32*3a42.546Z~start~'2023-12-05T19*3a29*3a54.229Z~timeType~'ABSOLUTE~tz~'LOCAL~editorString~'fields*20*40timestamp*2c*20*40message*0a*23*20*7c*20filter*20*40message*20like*20*2f9cb83be7-2d48-48d7-b852-64bab0fd32ea*2f*0a*7c*20filter*20*40requestId*20*3d*20*2701f39a7c-82bc-42a7-b59a-79d42af26994*27*0a*7c*20sort*20*40timestamp*20asc*0a*7c*20limit*20200~queryId~'47cf579de0b22d84-35cdceb0-4d9d170-df5fba6c-7ffea52bbeedc4bd84387~source~(~'*2faws*2flambda*2fazul-service-anvilprod))) Logs: 

```
[
    {
        "@timestamp": "2023-12-05 19:31:19.963",
        "@message": "START RequestId: 01f39a7c-82bc-42a7-b59a-79d42af26994 Version: $LATEST\n"
    },
    {
        "@timestamp": "2023-12-05 19:31:19.963",
        "@message": "[INFO]\t2023-12-05T19:31:19.963Z\t01f39a7c-82bc-42a7-b59a-79d42af26994\tazul.chalice\tReceived GET request for '/fetch/repository/files/9cb83be7-2d48-48d7-b852-64bab0fd32ea', with {\"query\": {\"catalog\": \"anvil3-it\", \"version\": \"2022-06-01T00:00:00.000000Z\"}, \"headers\": {\"accept-encoding\": \"identity\", \"cloudfront-forwarded-proto\": \"https\", \"cloudfront-is-desktop-viewer\": \"true\", \"cloudfront-is-mobile-viewer\": \"false\", \"cloudfront-is-smarttv-viewer\": \"false\", \"cloudfront-is-tablet-viewer\": \"false\", \"cloudfront-viewer-asn\": \"14618\", \"cloudfront-viewer-country\": \"US\", \"host\": \"service.prod.anvil.gi.ucsc.edu\", \"user-agent\": \"python-urllib3/1.26.18\", \"via\": \"1.1 4db130e87be66fce9731567ae0669c56.cloudfront.net (CloudFront)\", \"x-amz-cf-id\": \"gkjCdCFrddqOkAxJfHcHf6RryphoVHXf6aOZHrjeu_PCj35hETGnjw==\", \"x-amzn-trace-id\": \"Root=1-656f7a87-35d76ae162dcbdf777e90273\", \"x-forwarded-for\": \"35.168.152.160, 130.176.98.74\", \"x-forwarded-port\": \"443\", \"x-forwarded-proto\": \"https\"}}."
    },
    …
    {
        "@timestamp": "2023-12-05 19:31:20.265",
        "@message": "[INFO]\t2023-12-05T19:31:20.265Z\t01f39a7c-82bc-42a7-b59a-79d42af26994\tazul.terra\tMaking GET request to 'https://data.terra.bio/ga4gh/drs/v1/objects/v2_5f039240-868e-3d93-a4ff-bc9aaf98788b'"
    },
    {
        "@timestamp": "2023-12-05 19:31:20.265",
        "@message": "[DEBUG]\t2023-12-05T19:31:20.265Z\t01f39a7c-82bc-42a7-b59a-79d42af26994\tazul.terra\t… without request body"
    },
    {
        "@timestamp": "2023-12-05 19:31:20.269",
        "@message": "[ERROR]	2023-12-05T19:31:20.268Z	01f39a7c-82bc-42a7-b59a-79d42af26994	azul-service-anvilprod	Caught exception for path /fetch/repository/files/{file_uuid}
Traceback (most recent call last):
  File "/var/task/chalice/app.py", line 1914, in _get_view_function_response
    response = view_function(**function_args)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/app.py", line 1724, in fetch_repository_files
    body = _repository_files(file_uuid, fetch=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/app.py", line 1770, in _repository_files
    return app.repository_controller.download_file(catalog=catalog,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_controller.py", line 299, in download_file
    assert download.drs_uri is None, download
           ^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: TDRFileDownload(file_uuid='9cb83be7-2d48-48d7-b852-64bab0fd32ea', file_name='chr11.85200001_85300000.genotyped.vcf.gz.csi', file_version='2022-06-01T00:00:00.000000Z', drs_uri='drs://drs.anv0:v2_5f039240-868e-3d93-a4ff-bc9aaf98788b', replica=None, token=None)"
    },
    {
        "@timestamp": "2023-12-05 19:31:20.270",
        "@message": "[DEBUG]\t2023-12-05T19:31:20.270Z\t01f39a7c-82bc-42a7-b59a-79d42af26994\tazul.chalice\tReturning 500 response with headers {\"Content-Type\": \"text/plain\", \"Access-Control-Allow-Origin\": \"*\", \"Access-Control-Allow-Headers\": \"Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key\", \"Strict-Transport-Security\": \"max-age=31536000; includeSubDomains\"}. See next line for the first 1024 characters of the body.\nTraceback (most recent call last):\n  File \"/var/task/chalice/app.py\", line 1914, in _get_view_function_response\n    response = view_function(**function_args)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/var/task/app.py\", line 1724, in fetch_repository_files\n    body = _repository_files(file_uuid, fetch=True)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/var/task/app.py\", line 1770, in _repository_files\n    return app.repository_controller.download_file(catalog=catalog,\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/var/task/azul/service/repository_controller.py\", line 299, in download_file\n    assert download.drs_uri is None, download\n           ^^^^^^^^^^^^^^^^^^^^^^^^\nAssertionError: TDRFileDownload(file_uuid='9cb83be7-2d48-48d7-b852-64bab0fd32ea', file_name='chr11.85200001_85300000.genotyped.vcf.gz.csi', file_version='2022-06-01T00:00:00.000000Z', drs_uri='drs://drs.anv0:v2_5f039240-868e-3d93-a4ff-bc9aaf98788b', replica=None, token=None)\n"
    },
    {
        "@timestamp": "2023-12-05 19:31:20.271",
        "@message": "END RequestId: 01f39a7c-82bc-42a7-b59a-79d42af26994\n"
    },
    {
        "@timestamp": "2023-12-05 19:31:20.271",
        "@message": "REPORT RequestId: 01f39a7c-82bc-42a7-b59a-79d42af26994\tDuration: 308.44 ms\tBilled Duration: 309 ms\tMemory Size: 2048 MB\tMax Memory Used: 210 MB\t\n"
    }
]
```

I tried the same URL in the browser, swapping `anvil3-it` for `anvil3` and [initially](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2023-12-05T21*3a05*3a32.763Z~start~'2023-12-05T21*3a02*3a05.639Z~timeType~'ABSOLUTE~tz~'LOCAL~editorString~'fields*20*40timestamp*2c*20*40message*0a*23*20*7c*20filter*20*40message*20like*20*2f9cb83be7-2d48-48d7-b852-64bab0fd32ea*2f*0a*7c*20filter*20*40requestId*20*3d*20*271a8b833f-78e3-40c6-96a5-dbf897e0043c*27*0a*23*20*7c*20filter*20*40requestId*20*3d*20*27ee218a91-0aa5-4512-aa6e-31e7577a1335*27*0a*7c*20sort*20*40timestamp*20asc*0a*7c*20limit*20200~queryId~'47cf579de0b22d84-35cdceb0-4d9d170-df5fba6c-7ffea52bbeedc4bd84387~source~(~'*2faws*2flambda*2fazul-service-anvilprod))) received the same error, however upon retry the request [completed successfully](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2023-12-05T21*3a05*3a32.763Z~start~'2023-12-05T21*3a02*3a05.639Z~timeType~'ABSOLUTE~tz~'LOCAL~editorString~'fields*20*40timestamp*2c*20*40message*0a*23*20*7c*20filter*20*40message*20like*20*2f9cb83be7-2d48-48d7-b852-64bab0fd32ea*2f*0a*23*20*7c*20filter*20*40requestId*20*3d*20*271a8b833f-78e3-40c6-96a5-dbf897e0043c*27*0a*7c*20filter*20*40requestId*20*3d*20*27ee218a91-0aa5-4512-aa6e-31e7577a1335*27*0a*7c*20sort*20*40timestamp*20asc*0a*7c*20limit*20200~queryId~'47cf579de0b22d84-35cdceb0-4d9d170-df5fba6c-7ffea52bbeedc4bd84387~source~(~'*2faws*2flambda*2fazul-service-anvilprod))) and I was given a 302 response.

----

- [x] Security design review completed; the Resolution of this issue does **not** …
  - [x] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [x] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [x] … affect the permissions of external users like access to
    - TDR snapshots
  - [x] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [x] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [x] … affect monitoring of the system
  - [x] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [x] … add an interface that exposes sensitive or confidential data at the security boundary
  - [x] … affect the encryption of data at rest
  - [x] … require persistence of sensitive or confidential data that might require encryption at rest
  - [x] … require unencrypted transmission of data within the security boundary
  - [x] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [x] Documentation on any unchecked boxes is provided in comments below

**Comment by dsotirho-ucsc** [comment]
Similar failure (for a different file) on the 2nd attempt of IT.

IT Job [17125](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/17125):

```
2023-12-05 20:38:01,107    INFO MainThread test.integration_test: Beginning sub-test [repository_files] {'catalog': 'anvil3-it'}
2023-12-05 20:38:01,108    INFO MainThread test.integration_test: Making GET request to 'https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/f61d2014-fedd-4b5d-b75f-63c042c4a983?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z'
2023-12-05 20:38:01,109   DEBUG MainThread test.integration_test: … without request body
2023-12-05 20:38:01,467    INFO MainThread test.integration_test: Got 500 response after 0.358s from GET to https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/f61d2014-fedd-4b5d-b75f-63c042c4a983?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z
2023-12-05 20:38:01,467   DEBUG MainThread test.integration_test: … with response headers HTTPHeaderDict({'Content-Type': 'text/plain', 'Content-Length': '996', 'Connection': 'keep-alive', 'Date': 'Tue, 05 Dec 2023 20:38:01 GMT', 'x-amzn-RequestId': '74692bf5-ac73-4507-9831-f46dbe6c95e2', 'Access-Control-Allow-Origin': '*', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Access-Control-Allow-Headers': 'Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key', 'x-amz-apigw-id': 'PfKGeE0boAMEbwA=', 'X-Amzn-Trace-Id': 'Root=1-656f8a29-29a70cb97faf0e063ab38c82;Sampled=0;lineage=45061563:0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 de2ed3c94563fee614f35f9bc3f52d1c.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'IAD89-C3', 'X-Amz-Cf-Id': '9_6Hv8MvWPyuFlt-FI6Zc1-D9dCnZ6ZNfCiPsM64fNxpmhLSDEG51g=='})
2023-12-05 20:38:01,467   DEBUG MainThread test.integration_test: … with response body b'Traceback (most recent call last):\n  File "/var/task/chalice/app.py", line 1914, in _get_view_function_response\n    response ...'
2023-12-05 20:38:01,467    INFO MainThread test.integration_test: Failed sub-test [repository_files] {'catalog': 'anvil3-it'}
  test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [repository_files] (catalog='anvil3-it') ... FAIL
```  

[CloudWatch](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2023-12-05T20*3a39*3a49.737Z~start~'2023-12-05T20*3a35*3a59.146Z~timeType~'ABSOLUTE~tz~'LOCAL~editorString~'fields*20*40timestamp*2c*20*40message*0a*23*20*7c*20filter*20*40message*20like*20*2ff61d2014-fedd-4b5d-b75f-63c042c4a983*2f*0a*7c*20filter*20*40requestId*20*3d*20*27f1582198-28aa-4813-9024-e2f5a475840d*27*0a*7c*20sort*20*40timestamp*20asc*0a*7c*20limit*20200~queryId~'47cf579de0b22d84-35cdceb0-4d9d170-df5fba6c-7ffea52bbeedc4bd84387~source~(~'*2faws*2flambda*2fazul-service-anvilprod))) Logs:

```
[ERROR]	2023-12-05T20:38:01.453Z	f1582198-28aa-4813-9024-e2f5a475840d	azul-service-anvilprod	Caught exception for path /fetch/repository/files/{file_uuid}
Traceback (most recent call last):
  File "/var/task/chalice/app.py", line 1914, in _get_view_function_response
    response = view_function(**function_args)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/app.py", line 1724, in fetch_repository_files
    body = _repository_files(file_uuid, fetch=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/app.py", line 1770, in _repository_files
    return app.repository_controller.download_file(catalog=catalog,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_controller.py", line 299, in download_file
    assert download.drs_uri is None, download
           ^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: TDRFileDownload(file_uuid='f61d2014-fedd-4b5d-b75f-63c042c4a983', file_name='dchr18.PUR.counts.ancestry.txt', file_version='2022-06-01T00:00:00.000000Z', drs_uri='drs://drs.anv0:v2_e3986286-0f14-3527-86a2-d5b916fcb721', replica=None, token=None)
```

**Comment by dsotirho-ucsc** [comment]
On [3rd attempt](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/17128) on the IT run, similar `repository_files` error plus a `503 Service Unavailable` failure in `managed_access_repository_sources`

```
2023-12-05 21:24:35,058    INFO MainThread test.integration_test: Beginning sub-test [repository_files] {'catalog': 'anvil3-it'}
2023-12-05 21:24:35,059    INFO MainThread test.integration_test: Making GET request to '[https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/e0fb165d-9a81-4ea2-a959-f0aeeec81179?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z'](https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/e0fb165d-9a81-4ea2-a959-f0aeeec81179?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z%27)
2023-12-05 21:24:35,059   DEBUG MainThread test.integration_test: … without request body
2023-12-05 21:24:45,646    INFO MainThread test.integration_test: Got 500 response after 10.588s from GET to https://service.prod.anvil.gi.ucsc.edu/fetch/repository/files/e0fb165d-9a81-4ea2-a959-f0aeeec81179?catalog=anvil3-it&version=2022-06-01T00%3A00%3A00.000000Z
2023-12-05 21:24:45,647   DEBUG MainThread test.integration_test: … with response headers HTTPHeaderDict({'Content-Type': 'text/plain', 'Content-Length': '[101](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/17128#L101)1', 'Connection': 'keep-alive', 'Date': 'Tue, 05 Dec 2023 21:24:45 GMT', 'x-amzn-RequestId': '9599c4d0-7e5a-4e7b-9f08-733aa27de356', 'Access-Control-Allow-Origin': '*', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Access-Control-Allow-Headers': 'Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key', 'x-amz-apigw-id': 'PfQ7BEF-oAMEpIw=', 'X-Amzn-Trace-Id': 'Root=1-656f9513-5344afe76974b8946d0fa3a9;Sampled=0;lineage=45061563:0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 c35f767218cbd1125d801b52fa785c8c.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'IAD89-C3', 'X-Amz-Cf-Id': 'kVZhV043FJ3U-_B7rGEyMyhvwtq5pB2_OlZ4w__ba6j77IUYJAVlRw=='})
2023-12-05 21:24:45,647   DEBUG MainThread test.integration_test: … with response body b'Traceback (most recent call last):\n  File "/var/task/chalice/app.py", line 1914, in _get_view_function_response\n    response ...'
2023-12-05 21:24:45,647    INFO MainThread test.integration_test: Failed sub-test [repository_files] {'catalog': 'anvil3-it'}
  test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [repository_files] (catalog='anvil3-it') ... FAIL
```

```
FAIL: test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [managed_access_repository_sources] (catalog='anvil3-it')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 411, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 1268, in _test_managed_access
    public_source_ids = self._test_managed_access_repository_sources(catalog,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 1306, in _test_managed_access_repository_sources
    self.assertEqual(public_source_ids, list_source_ids())
                                        ^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 1289, in list_source_ids
    response = self._get_url_json(GET, url)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 784, in _get_url_json
    return json.loads(self._get_url_content(method, url))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 788, in _get_url_content
    response = self._get_url(method, url)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 806, in _get_url
    self._assertResponseStatus(response, 200, 301, 302)
  File "/builds/ucsc/azul/test/integration_test.py", line 847, in _assertResponseStatus
    assert response.status in [expected_status, *expected_statuses], (
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: (503, 'Service Unavailable', b'{"Code":"ServiceUnavailableError","Message":"No response from https://data.terra.bio/api/repository/v1/snapshots?offset=0&limit=1000&sort=created_date&direction=asc&filter=ANVIL_ within 5 seconds"}')
```

**Comment by dsotirho-ucsc** [comment]
```
Index: src/azul/service/repository_controller.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/service/repository_controller.py b/src/azul/service/repository_controller.py
--- a/src/azul/service/repository_controller.py	(revision 55c81bd1f235c2d61a4a26398facf7b18f783a14)
+++ b/src/azul/service/repository_controller.py	(date 1701814328935)
@@ -241,7 +241,7 @@
         try:
             download.update(plugin, authentication)
         except LimitedTimeoutException as e:
-            ServiceUnavailableError(*e.args)
+            raise ServiceUnavailableError(*e.args)
         if download.retry_after is not None:
             retry_after = min(download.retry_after, int(1.3 ** request_index))
             query_params = {

```

**Comment by dsotirho-ucsc** [comment]
Commit 0bd9865 on `develop` has been deployed:
- GitLab [dev](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/pipelines/28839)
- GitLab [anvildev](https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/pipelines/12201)
- GitLab [anvilprod](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/pipelines/7167)

**Comment by hannes-ucsc** [comment]
I don't think that commit fixes this issue. I must have made a typo in the title. The fix is 58556e09 and the backport is 2d3b4e29.
